course/src/main/java/org/openedx/course/presentation/info/CourseInfoFragment.kt

```diff
@@ -0,0 +1,304 @@
+                    if (showAlert) {
+                        InfoDialogFragment.newInstance(
+                            title = context.getString(R.string.course_enrollment_error),
+                            message = context.getString(
+                                R.string.course_enrollment_error_message,
+                                getString(CoreR.string.platform_name)
+                            )
+                        ).show(
+                            requireActivity().supportFragmentManager,
+                            InfoDialogFragment::class.simpleName
+                        )
+                    }
+                }
+
+                LaunchedEffect(enrollmentSuccess) {
+                    if (enrollmentSuccess.isNotEmpty()) {
+                        viewModel.onSuccessfulCourseEnrollment(
+                            fragmentManager = requireActivity().supportFragmentManager,
+                            courseId = enrollmentSuccess,
+                        )
+                    }
+                }
+
+                CourseInfoScreen(
+                    windowSize = windowSize,
+                    uiMessage = uiMessage,
+                    contentUrl = getInitialUrl(),
+                    uriScheme = viewModel.webViewConfig.uriScheme,
+                    hasInternetConnection = hasInternetConnection,
+                    checkInternetConnection = {
+                        hasInternetConnection = viewModel.hasInternetConnection
+                    },
+                    onBackClick = {
+                        requireActivity().supportFragmentManager.popBackStackImmediate()
+                    },
+                    onEnrollClick = { courseId ->
+                        viewModel.enrollInACourse(courseId)
+                    },
+                    refreshSessionCookie = {
+                        viewModel.tryToRefreshSessionCookie()
+                    },
+                    openExternalLink = { url ->
+                        ActionDialogFragment.newInstance(
+                            title = getString(CoreR.string.core_leaving_the_app),
+                            message = getString(
+                                CoreR.string.core_leaving_the_app_message,
+                                getString(CoreR.string.platform_name)
+                            ),
+                            url = url,
+                        ).show(
+                            requireActivity().supportFragmentManager,
+                            ActionDialogFragment::class.simpleName
+                        )
+                    },
+                    onInfoCardClicked = { pathId, infoType ->
+                        viewModel.infoCardClicked(
+                            fragmentManager = requireActivity().supportFragmentManager,
+                            pathId = pathId,
+                            infoType = infoType
+                        )
+                    },
```

Comment:
use case `onInfoCardClicked` here?

Reply:
The Course Info Webview includes information about Programs, and Programs, in turn, encompass data about Courses, creating a loop. This function is designed to provide support for these interrelated elements.